### PR TITLE
[AAQ-346] Switch from Nginx/Certbot to Caddy

### PR DESCRIPTION
**Pull request by sidravi1**
Reviewer: @amiraliemami 
Estimate: 30 mins

---

## Ticket

Fixes: [AAQ-346](https://idinsight.atlassian.net/browse/AAQ-346)

### Goal

Simplify arch

### Changes

Removed Nginx and Certbot
Added Caddy

## How has this been tested?

Run locally using docker compose.
Have not tested deployment on server!

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [x] I have updated the automated tests (if applicable)
- [x] I have updated the requirements (if applicable)
- [x] I have updated the README file (if applicable)
- [ x] I have updated affected documentation (if applicable)

[AAQ-346]: https://idinsight.atlassian.net/browse/AAQ-346?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/blog/posts/nginx-to-caddy.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+date: 2024-02-09
+---
+# Nginx out, Caddy in
+
+By swapping out Nginx for Caddy, we substantially simplified the deployment steps
+and the architecture - which means fewer docker containers to run and manage.
+
+<!-- more -->
+
+Previously, we were using Nginx and then manually running a script to issue certificates
```
Comment (amiraliemami) [inline docs/blog/posts/nginx-to-caddy.md:17]:
```suggestion
Previously, we were using NGINX and then manually running a script to issue certificates
```

Comment (amiraliemami) [inline docs/blog/posts/nginx-to-caddy.md:17]:
```suggestion
Previously, we were using NGINX and then manually running a script to issue certificates
```

File: docs/blog/posts/nginx-to-caddy.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+certificates for it. So we had to come up with a different process for local dev
+where we were issuing self-signed certs.
+
+## Welcome Caddy
+
+Oscar, our Google.org AI advisor's first advice when he saw our architecture was to switch
+to Caddy. Here are the benefits:
+
+1. It requests and refresh certs from Let's Encrypt.
+2. If your domain is localhost, it knows to issue it's own certificate.
```
Comment (amiraliemami) [inline docs/blog/posts/nginx-to-caddy.md:34]:
```suggestion
2. If your domain is localhost, it knows to issue its own certificate.
```

**Review by amiraliemami — COMMENTED** [review]
We should also remove the `nginx` folder at root, no? Or is that currently used for deployment?

File: deployment/Caddyfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{$DOMAIN:localhost} {
```
Comment (amiraliemami) [inline deployment/Caddyfile:1]: Cool default value behaviour!!

**Review by amiraliemami — CHANGES_REQUESTED** [review]
Runs locally thanks - so clean! 

Two changes:
- `nginx` root folder need to be deleted
- `deployment/template.env` needs to be updated

Also, should we test and add server deployment compatibility before we merge with main so we don't break our deployments?

**Comment by sidravi1** [comment]
> * `nginx` root folder need to be deleted

This is being used for deployment. I'll leave it for now till @yvonnegitau and @suzinyou look at deployment for `caddy`.

> * `deployment/template.env` needs to be updated

I don't think this needs to change. I have deleted the `template.env.nginx` file.

> Also, should we test and add server deployment compatibility before we merge with main so we don't break our deployments?

I'll check with @suzinyou. Deployments will definitely break but we can fix it in another PR. I want to make sure it doesn't mess up EG demo though since we use that for other demos.

**Review by amiraliemami — APPROVED** [review]
lgtm!

**Pull request merged**